### PR TITLE
WS Code Improvements

### PR DESCRIPTION
**Pull request by oddestdan**
Improve WS-related back-end and database structure, create string constants, simplify Mongo queries

### CODE REVIEW

File: socket/constants/answers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const USER_DISCONNECTED = 'User disconnected';
+const GAME_STARTED = 'Game started';
+const NEW_CHAT_MESSAGE = 'New chat message';
```
Comment (maxonchil) [inline socket/constants/answers.js:3]: If another one is called error, that one should be 'success'? Or Error logs and success logs

File: socket/index.js [hunk 2]
```diff
@@ -9,20 +9,24 @@ const {
...
-
             socket.on('create-room', async ({username, code: room}) => {
-                const roomWithUser = await roomsWhereUser(socket.id, username);
-                if (roomWithUser.length) {
-                    return checkForReconnection(socket, roomWithUser, room);
+                // Check if user is already present in any other room
+                const roomsWithUser = await roomsWhereUser(socket.id, username);
+                if (roomsWithUser.length) {
+                    // Check if user is trying to reconnect to same room
+                    checkForReconnection(socket, roomsWithUser.pop(), room);
```
Comment (maxonchil) [inline socket/index.js:27]: you need return here to stop the function working

File: socket/utilits/getUserBySocketId.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const getUserBySocketId = (room, socketId) => {
+  return room.users.find(user => user.socketId === socketId);
```
Comment (maxonchil) [inline socket/utilits/getUserBySocketId.js:2]: here is returning an array, but as i guess we waiting for string at FE

File: socket/utilits/getUsersInRoom.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 const getUsersInRoom = (room) => {
-    return room.users.map(item => Object.values(item).pop());
+    return room.users.map(user => user.username);
```
Comment (maxonchil) [inline socket/utilits/getUsersInRoom.js:2]: users in room should return an array of users in room

File: socket/utilits/getUserBySocketId.js (hunk 3 above)
Comment (oddestdan) [reply socket/utilits/getUserBySocketId.js:2]: this function is purely for use in BE (actually is only used once, but still)

File: socket/utilits/getUsersInRoom.js (hunk 4 above)
Comment (oddestdan) [reply socket/utilits/getUsersInRoom.js:2]: should've been renamed to getUsernamesInRoom from the start, probably

**Pull request merged**